### PR TITLE
RoR Group Project

### PR DESCRIPTION
**Pull request by firepicaso**
### Activities of this branch

- Used Ruby gems as a software packages system.
- Installed Ruby on Rails framework.
- Used controllers to handle requests and render empty views.
- Used params from browser requests in a safe way.
- Used preprocessed HTML file with embedded Ruby code.
- Used layouts and templates for shared content.
- Used database migration files to maintain database schema.
- Used validations for models.
- Secured app from n+1 problems.
- Set up associations between models.
- Built a web app that requires the user to log in.
- Used Devise gem for authentication.
- Limited access to web app resources based on authorization rules.
- Wrote unit tests using Rspec
- Wrote integration tests with capybara gem
- Fixed linter errors

kindly asking for a review

### CODE REVIEW

File: app/views/public_recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+              <p>Total food items: <%= recipe.recipe_foods.length %> </p>
+              <p>Total price: $<%= Recipe.calculate_total_amount(recipe) %></p>
+            </div>
+          </div>
+        <% end %>
+      </li>
+    <% end %>
+  </ul>
+  <% end %>
+</section>
```
Comment (Munsa1) [inline app/views/public_recipes/index.html.erb:25]: - [ ] Clicking on a public recipe should lead to the recipe show page even if the User clicking on the recipe did not create it. Currently, this brings about an error, kindly fix this.

Video Demo
[cinnamon-2023-11-26T000332+0200.webm](https://github.com/firepicaso/ror-recipe-app/assets/81932478/9700827c-32e1-4b34-9910-a6d1c689f03d)

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            <a id="openRecipeModal">Generate shopping list</a>
+            <a id="openInventoryModal">Add ingredient</a>
+        </div>
+      </div>
+    </div>
+  </div>
+
+  <% if @recipe.recipe_foods.empty? %>
+    <p>No ingredients added yet.</p>
+  <% else %>
```
Comment (Munsa1) [inline app/views/recipes/show.html.erb:26]: - [ ] Clicking on "Add ingredient" should lead to the form that allows the user to add new food, if the user is the owner of the recipe. Currently, this is not the case, rather the form for selecting Inventory items pops up. Kindly fix this.


Video Cast
[cinnamon-2023-11-26T001731+0200.webm](https://github.com/firepicaso/ror-recipe-app/assets/81932478/444298a8-d417-492f-a9b2-b16065849c7b)


#### Note that you need to swap the functionalities of the 2 buttons. clicking on ` Add Ingredient` should show the `Add recipe food` popup and clicking the `Generate shopping list` button should show the inventory pop up.

File: spec/integration/food_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @food1 = @user1.foods.create(food: 'Apple', measurement_unit: 'kg', price: 15)
+    @food2 = @user1.foods.create(food: 'Banana', measurement_unit: 'each', price: 10)
+
+    visit '/foods'
+  end
+
+  it 'should have the greetings' do
+    expect(page).to have_content('Food List')
+    expect(page).to have_link('Foods')
+    expect(page).to have_link('Recipes')
```
Comment (Munsa1) [inline spec/integration/food_index_spec.rb:22]: - [ ] Kindly fix some failing tests, as tests help ensure that your code works as expected

image
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @firepicaso @mumbereh @albertkantwi , 
required changes

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by joel-kalema — COMMENTED** [review]
Hi Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request  🛑 

pirates-of-the-caribbean-johnny-depp


- [ ] You submitted a project that doesn't have any commit about the changes required by the previous reviewer. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @joel-kalema in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**